include/picongpu/particles/Manipulate.hpp

```diff
@@ -20,21 +20,46 @@
 #pragma once
 
 #include "picongpu/simulation_defines.hpp"
-#include "picongpu/particles/filter/filter.def"
 #include "picongpu/particles/manipulators/manipulators.def"
+#include "picongpu/particles/filter/filter.def"
 
-#include <pmacc/Environment.hpp>
+#include <pmacc/particles/algorithm/CallForEach.hpp>
 #include <pmacc/meta/conversion/ToSeq.hpp>
 #include <pmacc/meta/ForEach.hpp>
 #include <pmacc/particles/meta/FindByNameOrType.hpp>
 
-#include <boost/mpl/apply.hpp>
-
 
 namespace picongpu
 {
 namespace particles
 {
+namespace detail
+{
+    /** Operator to create a filtered functor
+     */
+    template<
+        typename T_Filter,
+        typename T_Manipulator,
+        typename T_SpeciesType
+    >
```

Comment:
Not that important: why is the parameter order here different from `Manipulate`?

Reply:
I will change it to be consistent within the file